src/app/ui-library/input.tsx

```diff
@@ -0,0 +1,238 @@
+  color: ${(props) => props.theme.colors.elevation_n1};
+  border-radius: 50%;
+  margin-right: ${(props) => (props.$hasSiblings ? props.theme.spacing(4) : 0)}px;
+
+  :before,
+  :after {
+    position: absolute;
+    left: 7px;
+    content: ' ';
+    height: 8px;
+    width: 2px;
+    background-color: black;
+  }
+  :before {
+    transform: rotate(45deg);
+  }
+  :after {
+    transform: rotate(-45deg);
+  }
+`;
+
+const SubText = styled.div`
+  color: ${(props) => props.theme.colors.white_400};
+  margin-top: ${(props) => props.theme.spacing(4)}px;
+
+  ${(props) => props.theme.typography.body_s}
+`;
+
+const Feedback = styled.div`
+  margin-top: 8px;
+`;
+
+type Props = {
+  title?: string;
+  placeholder?: string;
+  value: string;
+  onChange: (event: ChangeEvent<HTMLInputElement>) => void;
+  onBlur?: (event: ChangeEvent<HTMLInputElement>) => void;
+  type?: 'text' | 'number';
+  hideClear?: boolean;
+  infoPanel?: React.ReactNode;
+  complications?: React.ReactNode;
+  variant?: InputVariant;
+  subText?: string;
+  className?: string;
+  disabled?: boolean;
+  feedback?: {
+    message: string;
+    variant?: FeedbackVariant;
+  }[];
+};
+
+function Input({
+  title,
+  placeholder,
+  value,
+  onChange,
+  onBlur,
+  type = 'text',
+  hideClear = false,
+  infoPanel,
+  complications,
+  variant = 'default',
+  subText,
+  className,
+  feedback,
+  disabled = false,
+}: Props) {
+  const inputRef = useRef<HTMLInputElement>(null);
+  const complicationsRef = useRef<HTMLDivElement>(null);
+
+  const handleChange: React.ChangeEventHandler<HTMLInputElement> = (e) => {
+    onChange(e);
+  };
+
+  useLayoutEffect(() => {
+    // This ensures that the complications don't overlap the text input
+    const stickyComplicationsRef = complicationsRef.current;
+    const stickyInputRef = inputRef.current;
+
+    if (stickyComplicationsRef && stickyInputRef) {
+      const padInput = () => {
+        const complicationsWidth = stickyComplicationsRef.clientWidth;
+        stickyInputRef.style.paddingRight = `${complicationsWidth + 16}px`;
+      };
+
+      padInput();
+
+      const resizeObserver = new ResizeObserver(padInput);
+      resizeObserver.observe(stickyComplicationsRef);
+
+      return () => {
+        resizeObserver.disconnect();
+      };
+    }
+  }, []);
+
+  const handleClear = () => {
+    onChange({ target: { value: '' } } as ChangeEvent<HTMLInputElement>);
+  };
+
+  return (
+    <Container className={className}>
+      {(title || infoPanel) && (
+        <TitleContainer>
+          <Title>{title}</Title>
+          {infoPanel && <InfoText>{infoPanel}</InfoText>}
+        </TitleContainer>
+      )}
+      <InputContainer>
+        <InputField
+          ref={inputRef}
+          type={type}
+          value={value}
+          onChange={handleChange}
+          onBlur={onBlur}
+          placeholder={placeholder}
+          disabled={disabled}
+          $variant={variant}
```

Comment:
consider using classNames here, so can use CSS class selectors

Reply:
Ah, that's a good idea 👍